src/coreclr/vm/ecalllist.h

```diff
@@ -598,6 +597,7 @@ FCFuncStart(gRuntimeHelpers)
     FCFuncElement("RegisterForGCReporting", GCReporting::Register)
     FCFuncElement("UnregisterForGCReporting", GCReporting::Unregister)
     FCFuncElement("Box", JIT_Box)
+    FCFuncElement("AllocateUninitializedObject", JIT_NewS_MP_FastPortable)
```

Comment:
This is correct in all cases. 
- There are special helpers for single-proc config. They have to be used when you are in the single proc config.
- arm32 needs to use a special helper that handles proper alignment.

There may be more corner cases that I am not thinking of at the moment.

I think you want to reuse the existing infra for allocating uninitialized objects as the other comment suggests.

Reply:
Ah, I see. Fixed in https://github.com/dotnet/runtime/pull/70181/commits/03b5b3315954c97c46dd431ce7c02cdaead92bdf with some guidance from Levi 🙂